app/components/Game.js

```diff
@@ -104,12 +104,11 @@ function ButtonGroup(props) {
 		<div  className = 'btnGrp'>
 				{
 					buttons.map((button, index) => {
-						const {text, classes, disabled, activated, onClick} = button;
+						const {text, classes, disabled, onClick} = button;
 						return (
 							<button
 								type = "button" 
-								aria-pressed = {activated()}
-								disabled = {disabled()}
+								disabled = {disabled}
 								key = {'button' + index} 
 								className = {classes}
 								style = {buttonHeight.style}
```

Comment:
A cleaner way to express props:
```js
const { style, } = buttonHeight
const buttonProps = {
  type: 'button',
  disabled,
  key: index,
  className: classes,
  style,
  onClick,
  children: text,
}
return <Button {...buttonProps} />
```
This is better in several ways:
1. Instead of using the ugly JSX key={{value}} verbose syntax, you use objects/classes/functions - with the full expressivity of the JS language
2. You separate props (in essence, they are function arguments) from the component structure, so you see your component tree at a glance.

Reply:
1. mind blown. so simple and somehow haven't crossed my mind so far.

2. about the key - you used `key: index`, which raised a question in my mind: I tried giving unique keys to all my elements that were constructed via `map()`, hence the combination of `'button' + index`, unlike `'cell' + y + x`. Does the fact that you used only `index` in your example mean that I can keep the keys not unique BETWEEN different maps?

Reply:
1. it's not that obvious. Did you read the react docs and official tutorial? ALL the docs?
2. https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
Reading now...

It's quite a lot, so I will be making some progress with the code cleanup as well.